modules/packages/Futures.chpl

```diff
@@ -99,11 +112,13 @@ module Futures {
 
     type retType;
 
-    var value: single retType;
     var valid: bool = false;
+    var value: retType;
+    var state: atomic bool;
```

Comment:
Re change from `single` to `atomic` and new field storing value; you could have switched to using a `single bool` instead of an atomic. Is there a reason to prefer the atomic? (I don't really have a preference; just trying to understand).